transport/http/client_test.go

```diff
@@ -98,8 +98,12 @@ func TestHTTPClient(t *testing.T) {
 }
 
 func TestHTTPClientBufferedStream(t *testing.T) {
+	// bodysize has a size big enought to make the resopnse.Body not an instant read
+	// so if the response is cancelled it wount be all readed and the test would fail
+	// The 6000 has not a particular meaning, it big enough to fulfill the usecase.
+	const bodysize = 6000
```

Comment:
I guess my question was more like: why is 6000 big enough, and 9000 too big? Is there a constant or default buffer size or something defined in the stdlib net/http package? If so, can you refer to that?

Reply:
I would also want to know that ... but I've tried to search for it, where the body is setted on a response and how is the `io.ReadCoser` is implemented on it, but It was impossible to find it for me, I would really like to have that answer because then I could provide a more accurate answer on what it's happening.

Also the 9000 it's not "too big" but it simply makes the `io.Read` works differently then. From the `io.Read`

> [...] An instance of this general case is that a Reader returning a non-zero number of bytes at the end of the input stream may return either err == EOF or err == nil. The next Read should return 0, EOF. [...]

So the next read to the body would return the `io.EOF` instead of the first one (I did try it while testing, but It's not something realted to `go-kit` but to the stdlib so IMO no test needed for this).

I think is on the way the body is read, I think that it's read by "chunks" from the "sender" instead of buffering all the body.